### PR TITLE
Extend Mach-O platform recognition

### PR DESCRIPTION
**Pull request by thestr4ng3r**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Some identical code from both mach-o 32 and 64 has been moved to
mach0_common.c to avoid duplicate compilation.
rz_mach0_platform_to_string() now recognizes all known platforms and the
information is shown as the subsys in the i command.
The "os" value was previously unreliably and ios sometimes showed as
"ios" and sometimes as "darwin" depending on the binary. Now the os is
"darwin" for all platforms. ios-* syscall files have thus been removed
as only darwin-* ones will be used.

**Test plan**

Open any mach-o and `i~^os` should display the actual platform of the binary (macOS, iOS, visionOS, etc.)

### CODE REVIEW

File: librz/bin/format/mach0/mach0.c [hunk 1]
```diff
@@ -3508,7 +3277,7 @@ void MACH0_(mach_headerfields)(RzBinFile *bf) {
 			if (!rz_buf_read_le32_at(buf, addr, &platform)) {
 				break;
 			}
-			cb_printf("0x%08" PFMT64x "  platform    %s\n", pvaddr, build_version_platform_to_string(platform));
+			cb_printf("0x%08" PFMT64x "  platform    %s\n", pvaddr, rz_mach0_platform_to_string(platform));
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0.c:3280]: We should get rid of cbprintf eventually

File: librz/bin/format/mach0/mach0_common.c [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	case CPU_TYPE_MC680x0:
+		return "mc680x0";
+	case CPU_TYPE_I386:
+	case CPU_TYPE_X86_64:
+		return "x86";
+	case CPU_TYPE_MC88000:
+		return "mc88000";
+	case CPU_TYPE_MC98000:
+		return "mc98000";
+	case CPU_TYPE_HPPA:
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_common.c:57]: I didn't knew it supported HPPA and SPARC

File: librz/bin/format/mach0/mach0_common.c [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+}
+
+/**
+ * String representation of CPU_SUBTYPE_* values as part of their \p cputype (CPU_TYPE_*)
+ */
+RZ_API RZ_NONNULL const char *rz_mach0_cpusubtype_tostring(ut32 cputype, ut32 cpusubtype) {
+	switch (cputype) {
+	case CPU_TYPE_VAX:
+		switch (cpusubtype) {
+		case CPU_SUBTYPE_VAX_ALL: return "all";
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_common.c:84]: And VAX

File: librz/bin/format/mach0/mach0_common.c [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+		case CPU_SUBTYPE_VAX730: return "vax730";
+		case CPU_SUBTYPE_UVAXI: return "uvaxI";
+		case CPU_SUBTYPE_UVAXII: return "uvaxII";
+		case CPU_SUBTYPE_VAX8200: return "vax8200";
+		case CPU_SUBTYPE_VAX8500: return "vax8500";
+		case CPU_SUBTYPE_VAX8600: return "vax8600";
+		case CPU_SUBTYPE_VAX8650: return "vax8650";
+		case CPU_SUBTYPE_VAX8800: return "vax8800";
+		case CPU_SUBTYPE_UVAXIII: return "uvaxIII";
+		default: return "Unknown vax subtype";
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_common.c]: Probably "VAX" here

File: librz/bin/format/mach0/mach0_common.c [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+		switch (cpusubtype & 0xff) {
+		case CPU_SUBTYPE_MC98000_ALL: return "all";
+		case CPU_SUBTYPE_MC98601: return "mc98601";
+		default: return "Unknown mc98000 subtype";
+		}
+	case CPU_TYPE_HPPA:
+		switch (cpusubtype & 0xff) {
+		case CPU_SUBTYPE_HPPA_7100: return "hppa7100";
+		case CPU_SUBTYPE_HPPA_7100LC: return "hppa7100LC";
+		default: return "Unknown hppa subtype";
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_common.c]: "HPPA" probably

**Review by XVilka — APPROVED** [review]

File: librz/bin/format/mach0/mach0_common.c (hunk 2 above)
Comment (thestr4ng3r) [reply librz/bin/format/mach0/mach0_common.c:57]: I think these still come from CMU Mach

File: librz/bin/format/mach0/mach0.c (hunk 1 above)
Comment (thestr4ng3r) [reply librz/bin/format/mach0/mach0.c:3280]: agreed

**Pull request merged**